Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -13276,6 +13225,43 @@ Ref<WebBackForwardList> WebPageProxy::protectedBackForwardList() const
     return m_backForwardList;
 }
 
+template<typename F>
+void WebPageProxy::sendToWebPage(const std::optional<FrameIdentifier>& frameID, F&& sendFunction)
+{
+    if (frameID) {
+        if (RefPtr frame = WebFrameProxy::webFrame(*frameID)) {
+            if (RefPtr remotePageProxy = frame->remotePageProxy()) {
+                sendFunction(remotePageProxy.get());
+                return;
+            }
+        } else {
+            ASSERT_NOT_REACHED();
+            return;
+        }
+    }
+    sendFunction(this);
+}
+
+template<typename M, typename C>
+void WebPageProxy::sendAsyncMessageToWebPage(const std::optional<FrameIdentifier>& frameID, M&& message, C&& completionHandler)
```

Comment:
I think all the callers of this have a non-optional FrameIdentifier.  Is that right?  Otherwise, this should probably not be a const reference because it can just be in registers so there's no reason to pass only a pointer.

Reply:
The calls in `dragEnded()` and `performDragControllerAction()` have an optional FrameIdentifier.